version: '3.9'
services:

  db:
    container_name: db
    image: postgres
    networks: 
      - iteen
    restart: always
    ports:
      - '5432:5432'
    environment:
      POSTGRES_USER: admin
      POSTGRES_PASSWORD: admin
      POSTGRES_DB: testbase
      POSTGRES_INITDB_ARGS: "-A md5"  
    volumes:
      - ./data:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U admin -d testbase"]
      interval: 10s
      timeout: 5s
      retries: 5
      start_period: 10s

  pgadmin:
    container_name: pgadmin_container
    image: dpage/pgadmin4
    environment:
      PGADMIN_DEFAULT_EMAIL: "admin@admin.com"
      PGADMIN_DEFAULT_PASSWORD: "admin"
      PGADMIN_CONFIG_SERVER_MODE: "False"
    volumes:
      - ./pgadmin:/var/lib/pgadmin
    ports:
      - "5050:80"
    restart: always
    networks:
      - iteen
    depends_on:
      - db

  webapp:
    build:
      args:
        - PIP_VERSION=${WEBAPP_PIP_VERSION:-23.1.2}
        - POETRY_VERSION=${WEBAPP_POETRY_VERSION:-1.4.2}
        - PYTHON_VERSION=${WEBAPP_PYTHON_VERSION:-3.11}
        - VERSION=${WEBAPP_VERSION:-dev}
      context: .
    container_name: webapp
    image: webapp:latest
    networks:
      - iteen
    ports:
      - "8000:8000"
    volumes:
      - ./:/app/:ro
    init: true
    depends_on:
      - db
      
networks:
  iteen:
    name: iteen.network
    driver: bridge
